apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1299 @@
+		PRNT("Error: Fail to init");
+		return -1;
+	}
+	response->message = malloc(WEBCLIENT_CONF_MAX_MESSAGE_SIZE);
+	if (response->message == NULL) {
+		PRNT("Error: Fail to init");
+		free(response->phrase);
+		return -1;
+	}
+	response->headers = malloc(sizeof(struct http_keyvalue_list_t));
+	if (response->headers == NULL || http_keyvalue_list_init(response->headers) < 0) {
+		PRNT("Error: Fail to init");
+		free(response->phrase);
+		free(response->message);
+		return -1;
+	}
+	FUNC_EX;
+	return 0;
+}
+
+void ws_test_http_client_response_release(struct http_client_response_t *response)
+{
+	FUNC_EN;
+	free(response->phrase);
+	free(response->message);
+	if (http_keyvalue_list_release(response->headers) != 0) {
+		PRNT("http_keyvalue_list_release(response->headers) failed");
+	}
+	free(response->headers);
+}
+
+static int ws_test_wget_gethostip(FAR char *hostname, in_addr_t *ipv4addr)
+{
+	FUNC_EN;
+#ifdef CONFIG_NET_LWIP_NETDB
+	FAR struct hostent *he;
+	he = gethostbyname(hostname);
+	if (he == NULL) {
+		PRNT("gethostbyname failed");
+		return -ENOENT;
+	} else if (he->h_addrtype != AF_INET) {
+		PRNT("gethostbyname returned an address of type: %d", he->h_addrtype);
+		return -ENOEXEC;
+	}
+
+	memcpy(ipv4addr, he->h_addr, sizeof(in_addr_t));
+	FUNC_EX;
+	return OK;
+#else
+	PRNT("NETDB is not supported");
+	FUNC_EX;
+	return ERROR;
+#endif
+}
+
+
+#ifdef CONFIG_NET_SECURITY_TLS
+static void ws_test_wget_tls_debug(void *ctx, int level, const char *file, int line, const char *str)
+{
+	printf("%s:%04d: %s", file, line, str);
+}
+
+static void ws_test_wget_tls_release(struct http_client_tls_t *client)
+{
+	FUNC_EN;
+	if (client == NULL) {
+		return;
+	}
+
+	mbedtls_x509_crt_free(&(client->tls_rootca));
+	mbedtls_x509_crt_free(&(client->tls_clicert));
+	mbedtls_pk_free(&(client->tls_pkey));
+	mbedtls_ssl_config_free(&(client->tls_conf));
+	mbedtls_ctr_drbg_free(&(client->tls_ctr_drbg));
+	mbedtls_entropy_free(&(client->tls_entropy));
+	mbedtls_ssl_session_free(&(client->tls_session));
+	FUNC_EX;
+}
+
+int ws_test_webclient_tls_init(struct http_client_tls_t *client, struct http_client_ssl_config_t *ssl_config)
+{
+	FUNC_EN;
+	int result = 0;
+
+	if (!client || !ssl_config) {
+		return -1;
+	}
+
+	mbedtls_ssl_config_init(&(client->tls_conf));
+	mbedtls_x509_crt_init(&(client->tls_rootca));
+	mbedtls_x509_crt_init(&(client->tls_clicert));
+	mbedtls_pk_init(&(client->tls_pkey));
+	mbedtls_entropy_init(&(client->tls_entropy));
+	mbedtls_ctr_drbg_init(&(client->tls_ctr_drbg));
+	mbedtls_ssl_session_init(&(client->tls_session));
+	mbedtls_ssl_conf_authmode(&(client->tls_conf), ssl_config->auth_mode);
+#ifdef MBEDTLS_DEBUG_C
+	mbedtls_debug_set_threshold(MBED_DEBUG_LEVEL);
+#endif
+
+	/* 0. Initialize the RNG */
+	PRNT("  . Seeding the random number generator...");
+
+	if ((result = mbedtls_ctr_drbg_seed(&(client->tls_ctr_drbg), mbedtls_entropy_func, &(client->tls_entropy), (const unsigned char *)tlsname, strlen(tlsname))) != 0) {
```

Comment:
The line is too long, please divide code to increase readability.

Reply:
Done